features/connection/src/main/java/com/kape/connection/ui/mobile/ConnectionScreen.kt

```diff
@@ -171,6 +186,66 @@ fun ConnectionScreen() = Screen {
                             state = state.value,
                         )
                     }
+                    showRatingGeneralDialog.value =
+                        state.value.ratingDialogType is RatingDialogType.General
+                    showRatingReviewDialog.value =
+                        state.value.ratingDialogType is RatingDialogType.Review
+                    showRatingFeedbackDialog.value =
+                        state.value.ratingDialogType is RatingDialogType.Feedback
+
+                    if (showRatingGeneralDialog.value) {
+                        RatingDialog(
+                            onConfirm = {
+                                viewModel.showReviewPrompt()
+                            },
+                            onDismiss = {
+                                viewModel.showFeedbackPrompt()
+                            },
+                        )
+                    }
+
+                    if (showRatingReviewDialog.value) {
+                        val url = stringResource(id = R.string.app_url)
+                        val context = LocalContext.current
+
+                        RatingReviewDialog(
+                            onConfirm = {
+                                val launchIntent = Intent(Intent.ACTION_VIEW)
+                                launchIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                                launchIntent.data = Uri.parse(url)
+
+                                // Silently fail if Google Play Store isn't installed.
+                                if (launchIntent.resolveActivity(context.packageManager) != null) {
+                                    context.startActivity(launchIntent)
+                                }
+                                viewModel.setRatingStateInactive()
+                            },
+                            onDismiss = {
+                                viewModel.setRatingStateInactive()
```

Comment:
I haven't tried to debug it, but this doesn't seem to be working as expected. I can't dismiss the dialog.

Reply:
Hmmm, I'll have a look. Thanks!

Reply:
Addressed.